src/Symfony/Component/Scheduler/Scheduler.php

```diff
@@ -62,8 +66,23 @@ public function run(array $options = []): void
 
             $ran = false;
             foreach ($this->generators as $generator) {
-                foreach ($generator->getMessages() as $message) {
+                foreach ($generator->getMessages() as $context => $message) {
+                    if ($this->dispatcher) {
```

Comment:
What about having testing `!$this->dispatcher` to have less code in the condition?